app/controllers/admin/settings/project_custom_fields_controller.rb

```diff
@@ -75,13 +78,18 @@ def project_mappings
     end
 
     def unlink
-      project = Project.find(permitted_params.project_custom_field_project_mapping[:project_id])
-      project_custom_field_mapping = @custom_field.project_custom_field_project_mappings.find_by(project:)
       delete_service = ProjectCustomFieldProjectMappings::DeleteService
-                         .new(user: current_user, model: project_custom_field_mapping)
+                         .new(user: current_user, model: @project_custom_field_mapping)
                          .call
 
-      delete_service.on_success { render_unlink_response(project:) }
+      delete_service.on_success { render_unlink_response(project: @project) }
+
+      delete_service.on_failure do
+        update_flash_message_via_turbo_stream(
+          message: join_flash_messages(delete_service.errors.full_messages),
+          full: true, dismiss_scheme: :hide, scheme: :danger
+        )
+      end
```

Comment:
In case of contract validation error: E.g. attempting to unlink a required custom field

![Screenshot 2024-05-22 at 4 20 51 PM](https://github.com/opf/openproject/assets/17295175/22f63d28-f277-4d30-9d44-bc9561ada8e1)